src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerSpec.scala

```diff
@@ -373,6 +371,74 @@ class JsonRpcControllerSpec
     response.result shouldBe Some(expectedTxResponse)
   }
 
+  it should "handle eth_getRawTransactionByBlockHashAndIndex request" in new TestSetup {
+    val blockToRequest = Block(Fixtures.Blocks.Block3125369.header, Fixtures.Blocks.Block3125369.body)
+    val txIndexToRequest = blockToRequest.body.transactionList.size / 2
+
+    blockchain.storeBlock(blockToRequest).commit()
+    blockchain.saveBestKnownBlock(blockToRequest.header.number)
+
+    val request: JsonRpcRequest = JsonRpcRequest(
+      "2.0",
+      "eth_getRawTransactionByBlockHashAndIndex",
+      Some(
+        JArray(
+          List(
+            JString(s"0x${blockToRequest.header.hashAsHexString}"),
+            JString(s"0x${Hex.toHexString(BigInt(txIndexToRequest).toByteArray)}")
+          )
+        )
+      ),
+      Some(JInt(1))
+    )
+    val response = jsonRpcController.handleRequest(request).futureValue
+    val expectedTxResponse = rawTrnHex(blockToRequest.body.transactionList, txIndexToRequest)
+
+    response.jsonrpc shouldBe "2.0"
+    response.id shouldBe JInt(1)
+    response.error shouldBe None
+    response.result shouldBe expectedTxResponse
+  }
+
+  it should "handle eth_getRawTransactionByHash request" in new TestSetup {
+    val mockEthService = mock[EthService]
+    override val jsonRpcController =
+      new JsonRpcController(
+        web3Service,
+        netService,
+        mockEthService,
+        personalService,
+        None,
+        debugService,
+        qaService,
+        config
+      )
+
+    val txResponse: SignedTransaction = Fixtures.Blocks.Block3125369.body.transactionList.head
+    (mockEthService.getRawTransactionByHash _)
+      .expects(*)
+      .returning(Future.successful(Right(RawTransactionResponse(Some(txResponse)))))
+
+    val request: JsonRpcRequest = JsonRpcRequest(
+      "2.0",
+      "eth_getRawTransactionByHash",
+      Some(
+        JArray(
+          List(
+            JString("0xe9b2d3e8a2bc996a1c7742de825fdae2466ae783ce53484304efffe304ff232d")
+          )
+        )
+      ),
+      Some(JInt(1))
+    )
+
+    val response = jsonRpcController.handleRequest(request).futureValue
```

Comment:
there is set of additional matchers for rpc responses in `io.iohk.ethereum.jsonrpc.JRCMatchers`, it allows to compare result only, so less boilerplate

Reply:
Custom marchers from JRCMatchers were used  to simplify tests: https://github.com/input-output-hk/mantis/pull/720/commits/7bb0d21bd3bfa3687c4ce31de059a7e54de71c6c